athenahealth/lab_results.go

```diff
@@ -128,20 +128,35 @@ const (
 	LabResultAttachmentTypeTIFF LabResultAttachmentType = "TIFF"
 )
 
+type observationDateTime struct {
+	t           time.Time
+	includeTime bool
+}
+
+// NewObservationDate sets `observationdate` & `observationtime`
+func NewObservationDateTime(t time.Time) *observationDateTime {
+	return &observationDateTime{t, true}
+}
+
+// NewObservationDate sets only `observationdate`
+func NewObservationDate(t time.Time) *observationDateTime {
+	return &observationDateTime{t, false}
+}
+
 type AddLabResultDocumentOptions struct {
 	// AttachmentContents must be Base64 encoded
-	AttachmentContents io.Reader               `json:"attachmentcontents"`
-	AttachmentType     LabResultAttachmentType `json:"attachmenttype"`
-	InternalNote       *string                 `json:"internalnote"`
-	NoteToPatient      *string                 `json:"notetopatient"`
-	ObservationDate    *time.Time              `json:"observationdate"`
-	OriginalFilename   *string                 `json:"originalfilename"`
+	AttachmentContents io.Reader
+	AttachmentType     LabResultAttachmentType
+	InternalNote       *string
+	NoteToPatient      *string
+	ObservedAt         *observationDateTime
```

Comment:
I wonder if `ObservationDateTime` would align better to the athena terms here

Reply:
agreed that seems more aligned